timestamp-client/src/main/java/com/palantir/timestamp/RateLimitedTimestampService.java

```diff
@@ -56,13 +62,29 @@
     private final AtomicReference</* nullable */ TimestampHolder> currentBatch =
             new AtomicReference<TimestampHolder>();
 
+    // Required for large internal product.
     public RateLimitedTimestampService(TimestampService delegate, long minTimeBetweenRequestsMillis) {
         this.delegate = TimingProxy.newProxyInstance(TimestampService.class, delegate, timer);
+        this.configSupplier = () -> ImmutableTimestampClientConfig.builder().enableTimestampBatching(true).build();
         this.minTimeBetweenRequestsMillis = minTimeBetweenRequestsMillis;
     }
 
+    // Note that calls to configSupplier *MUST* be cheap, because it is called on every timestamp request.
+    public RateLimitedTimestampService(TimestampService delegate, Supplier<TimestampClientConfig> configSupplier) {
```

Comment:
This class feels like it's doing two things now that it's accepting a config object - might be better to move the dispatching behavior into a layer above this. (this would also make testing easier)